src/render/render_quantum.rs

```diff
@@ -453,6 +453,12 @@ impl AudioRenderQuantum {
         self.channels.truncate(1);
     }
 
+    /// Check if this `RenderQuantum` contains a single channel of silence
+    #[must_use]
+    pub fn is_silent(&self) -> bool {
```

Comment:
Why require only one channel?

Reply:
It questioned me too, I'm not completely sure about this one. In https://webaudio.github.io/web-audio-api/#AudioNode-actively-processing, the spec says that: 

> `[AudioNode](https://webaudio.github.io/web-audio-api/#AudioNode)s that are not [actively processing](https://webaudio.github.io/web-audio-api/#actively-processing) output a single channel of silence.`. 

I have the impression (but I'm not sure) that will more simple to reason about in the long run.

Reply:
I currently do not see how this would relate to the spec text.
In any case, the current name is misleading and should probably be called `is_single_channel_of_silence`.
Also, I do not see any drawbacks in just implementing
```rust
for c in &self.channels {
    if !c.is_silent() { return false }
}
return true
```

Reply:
hum, my point in implementing `is_silence()` was really about the `actively processing` stuff. I really think we should consider that an `AudioRenderQuantum` containing several channel of silence is wrong (because of the spec, and because that would mean we do something, e.g. mixing, that is unnecessary), so I think we need a way to track if such problem appears. 
On the other end, I really agree with you that the implementation you propose is more safe. So what about something like (at least in a temporary way):

```rs
if cfg(test) {
   if one_channel_of_silence { 
      return true;
   } else if several_channels_of_silence {
      panic!("this should not happen...");
   }
} else {
    for c in &self.channels {
       if !c.is_silent() { return false; }
    }
    return true;
}
```

This way end-user applications are safe but we can write tests to ensure that the propagation of silence is properly done?

Reply:
I see your point, but since the function will be part of the public interface, I think we should stick to the principle of least surprise.
Switching behaviour between test and not-test could be very surprising to end users.
Also, currently, we allow to upmix a single channel of silence to multiple channels. We don't panic at that call site either.